app/src/main/res/layout/story_chapter_view.xml

```diff
@@ -89,10 +90,8 @@
             android:layout_marginTop="8dp"
             android:layout_marginEnd="8dp"
             android:layout_marginBottom="4dp"
-            android:ellipsize="end"
             android:fontFamily="sans-serif"
-            android:maxLines="4"
-            android:text="@{htmlContent}"
+            android:text="@{viewModel.chapterSummary.chapterPlayState != ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES ? htmlContent : String.format(@string/chapter_prerequisite, viewModel.index, viewModel.previousChapterTitle)}"
```

Comment:
Databinding has built-in formatting, so this can be simplified to:

```
@{viewModel.chapterSummary.chapterPlayState != ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES ? htmlContent : @string/chapter_prerequisite(viewModel.index, viewModel.previousChapterTitle)}
```

Reply:
Nice, thanks!